### PR TITLE
Add additional binary device classes to inversion list

### LINKED ISSUE #9239: The colors on some binary_sensors are inverted/misrepresented (red/green)
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When looking at an entity's _**history**_, some binary sensors are visually represented with the wrong or inverted color. 
For example with device class 'sound' or 'vibration', the state 'clear' shows red, while 'detected' shows green. (confusing)

On the other hand, device classes like 'door' and 'problem' work just like one would expect, showing green for closed/ok and red for open/problem. This is the way it should be.

image

### Describe the behavior you expected

I think the history colors of these device classes should be inverted, so that they actually represent the state that matches the intuitive expectation (like 'door' does), 
OR
even better, (if not too complicated/bloating) let the user **pick history colors for each state** of each entity. 

If that bloats the UI too much, maybe only add a simple yaml key to `customize.yaml` like
```
customize:
  binary_sensor.doorbell:
    device_class: sound
    invert_history_color: true
``` 
for example.

until then, a workaround could be to use the device class 'door' for a sensor that, in reality, monitors vibration.
### Steps to reproduce the issue

1. Add a binary_sensor of device class 'sound'
2. Change its state at least once
3. Look at the entity's history via **History** or a **History Graph card**.

### What version of Home Assistant Core has the issue?

core-2021.5.5

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Firefox 88.0.1

### Which operating system are you using to run this browser?

Windows 10

### State of relevant entities

```txt
binary_sensor.front_doorbell 	off 	
friendly_name: Front Doorbell
device_class: sound
```

### Problem-relevant frontend configuration

```yaml
(ESPHome yaml)
  - platform: gpio
    pin:
      number: D3
      mode: INPUT_PULLUP
      inverted: true
    name: "Front Doorbell"
    device_class: sound
    filters:
      - delayed_off: 3s
```

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

I went through the list https://www.home-assistant.io/integrations/binary_sensor/ and added the ones that I think should be inverted as well.

Not completely sure about `cold` and `heat` but since the page says `off` is normal, I lean towards inverting them as well.

I think their intention is to highlight when something is "too hot" or "too cold", so off the normal state. If that is true, then the inversion makes sense.

@matthiasdebaat Do you agree with the representation?

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #9239
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/chart/state-history-chart-timeline.ts [hunk 1]
```diff
@@ -17,15 +17,21 @@ import type { TimeLineData } from "./timeline-chart/const";
  */
 const BINARY_SENSOR_DEVICE_CLASS_COLOR_INVERTED = new Set([
```
Comment (bramkragten) [inline src/components/chart/state-history-chart-timeline.ts:18]: I wonder if we don't have more inverted than not now? Should we reverse the normal logic?

**Comment by matthiasdebaat** [comment]
> I went through the list https://www.home-assistant.io/integrations/binary_sensor/ and added the ones that I think should be inverted as well.
> 
> Not completely sure about `cold` and `heat` but since the page says `off` is normal, I lean towards inverting them as well.
> 
> I think their intention is to highlight when something is "too hot" or "too cold", so off the normal state. If that is true, then the inversion makes sense.
> 
> @matthiasdebaat Do you agree with the representation?

Good one!

File: src/components/chart/state-history-chart-timeline.ts (hunk 1 above)
Comment (spacegaier) [reply src/components/chart/state-history-chart-timeline.ts:18]: With this PR we would now have 17 of 26 device classes inverted.

=> Should I invert the inversion logic :D ?

Comment (bramkragten) [reply src/components/chart/state-history-chart-timeline.ts:18]: Yes please

**Review by bramkragten — APPROVED** [review]

**Pull request merged**